d1985r0.md

```diff
@@ -428,58 +427,88 @@ template<typename T, typename A> my_container(T*, size_t, const A&) -> my_contai
 
 As the primary template is not defined there is only a default
 constructor implicit deduction guide. The explicit deduction guides
-select the first specialization unless an allocator object is given
-in which case the second specialization is selected. To select the
-third specialization the template arguments must be explicitly given.
+select the first specialization unless an allocator object is given in
+which case the second specialization is selected. To select the third
+specialization the template arguments must be explicitly given.
 
 
-## Overloading of variable templates
+## Impacts on the specialization of variable templates
 
-Universal template parameters can be used to implement what appears like overloading a variable
-template. The problem of not being able to delete the base case then becomes more
-pressing than currently as selecting the base case should trigger an error but
-will be unconditionally selected when none of the specialziations matches. The dependent
-static_assert mentioned above can be used to rescue this situation but
-this is still somewhat of a hack.
+Universal template parameters can be used to implement what appears
+like overloading of variable templates. 
+
+The problem of not being able to delete the base case then becomes
+more pressing than currently as selecting the base case should trigger
+an error but will be selected when none of the specialziations
+matches. This is solved by [@p2041r0] which is assumed in the example
+below.
 
 ```cpp
+
+// Metafunction to find a tuple element by a type predicate.
+template<template<typename> class PRED, size_t POS, typename TL> 
+constexpr size_t tuple_find()
+{
+    if constexpr (POS == tuple_size<TL>()) {
+        return npos;
+    }
+    else if constexpr (PRED<remove_cvref_t<tuple_element_t<POS, TL>>>::value)
+        return POS;
+    else
+        return tuple_find<PRED, POS + 1, TL>();
+}
+template<template<typename> class PRED, typename TL> 
+constexpr size_t tuple_find() { return tuple_find<PRED, 0, TL>(); }
+
+// Helper to bind the first arguments of a provided template
+template<template<typename auto...> TPL, template auto... BS> struct {
```

Comment:
s/typename auto/template auto/
also I think you forgot to call this "curry" :)

Reply:
Right, fell into the same trap. Hard to find the bugs when you're not supposed to be able to compile,